code/game/machinery/bots/cprbot.dm

```diff
@@ -0,0 +1,509 @@
+
+		var/mob/living/carbon/human/another_bot_patient
+
+		// sanity checks
+		if (another_cpr_bot.human != null)
+			another_bot_patient = another_cpr_bot.human.resolve()
+
+		// sanity checks
+		if (another_bot_patient == null)
+			continue
+
+		// Another CPR bot is targetting this patient, skip
+		if (another_bot_patient in potential_patients)
+			potential_patients.Remove(another_bot_patient)
+
+	if (potential_patients.len)
+		var/mob/living/carbon/human/patient = potential_patients[1]
+		human = WEAKREF(patient)
+		if (patient && !(patient in has_said_to_patient))
+			visible_message("[patient] is injured! I'm coming!")
+			has_said_to_patient += patient
+
+		move_to_target()
+	else
+		// If no patients are found, check if owner is nearby and follow them if idle
+		if (state == STATE_CPRBOT_IDLE && (owner && (owner in view(search_radius, src))))
+			state = STATE_CPRBOT_FOLLOWING_OWNER
+			walk_to(src, owner, 0, movement_delay)
+		else if (state == STATE_CPRBOT_FOLLOWING_OWNER)
+			// Continue following the owner if no patient is in sight
+			walk_to(src, owner, 0, movement_delay)
+		else
+			go_idle()
+
+/obj/structure/machinery/bot/cprbot/proc/call_astar_pathfinding()
+	if (human == null)
+		return FALSE
+
+	var/mob/living/carbon/human/patient = human.resolve()
+	if (patient == null)
+		return FALSE
+
+	var/turf/target_turf = get_turf(patient)
+
+	if (!target_turf)
+		return
+
+	// Calculate the path using your A* function
+	path = AStar(src.loc, target_turf, /turf/proc/CardinalTurfsWithAccess, /turf/proc/Distance, 0, 30, id=botcard)
+
+	if (!length(path))
+		return
+
+	// Delete the straight line sections since walk_to moves better without that.
+	var/list/path_new = list()
+	var/turf/last = path[path.len]
+
+	if (!istype(last, /turf)) // Ensure last is a turf before accessing it
+		return
+
+	path_new.Add(path[1])
+
+	for (var/i = 2; i < path.len; i++) {
+		if (istype(path[i], /turf) && istype(path[i + 1], /turf)) {
+			var/turf/current_turf = path[i]
+			var/turf/next_turf = path[i + 1]
+
+			if ((next_turf.x == current_turf.x) || (next_turf.y == current_turf.y)) { // We have a straight line, scan for more to cut down
+				path_new.Add(current_turf)
+				for (var/j = i + 1; j < path.len; j++) {
+					if (istype(path[j + 1], /turf)) {
+						var/turf/next_next_turf = path[j + 1]
+						var/turf/prev_turf = path[j - 1]
+
+						// This is a corner and the endpoint of our line
+						if ((next_next_turf.x != prev_turf.x) && (next_next_turf.y != prev_turf.y)) {
+							path_new.Add(path[j])
+							i = j + 1
+							break
+						}
+					}
+
+					if (j == path.len - 1) {
+						path = list()
+						path = path_new.Copy()
+						path.Add(last)
+						return
+					}
+				}
+			} else {
+				path_new.Add(current_turf)
+			}
+		}
+	}
+
+	path = list()
+	path = path_new.Copy()
+	path.Add(last)
+
+	if (!path || length(path) == 0) {
+		src.human = null
+		src.currently_healing = FALSE
+		return
+	}
+
+	// Set the path
+	src.path = path
```

Comment:
are you setting the var to itself here? Why? If there is two different but same name variables, rename them